File: Java/Easy/Climbing Stairs.java
public class Solution {
    /**
     * @param n: An integer
     * @return: An integer
     */
    public int climbStairs(int n) {
        // write your code here
        if(n <= 0) {
            return 0;
        }
        if(n == 1) {
            return 1;
        }

        int sum = 0;
        int one = 0;
        int two = 1;
        for(int i = 1; i<=n; i++) {
            sum = one + two;
            one = two;
            two = sum;
        }
        return sum;
    }
}

File: C++/Easy/Climbing Stairs.cpp
class Solution {
public:
    /**
     * @param n: An integer
     * @return: An integer
     */
    int climbStairs(int n) {
        // write your code here
        if(n<=0) {
            return 0;
        }
        if(n == 1 || n == 2) {
            return n;
        }
        int one = 1, two = 2, ret = 3;
        for(int i=3; i<=n; i++) {
            ret = one + two;
            one = two;
            two = ret;
        }
        return ret;
    }
};
